### ISSUE TITLE
How to Save Properly Scaled Images from gg_genetracks Plot

### ISSUE THREAD

**Issue by Xi-Cao** [myles-lewis/locuszoomr#30]
Hi there, thanks for your helpful and convenient package on locuszoom plot！
I used the `gg_genetracks` function to create a gene position plot. However, when I attempted to save the plot with `ggsave` function, I noticed that the sizes of the genes and texts did not scale as expected. I suspect this might be due to the modifications I made for adding the axis.
I observed that the (if (!is.null(g)) gggrid::grid_panel(g)) part of my plot does not align with the axis in the saved PDF. Could you please help me resolve this issue?

Here is my code:
```
gene_plot <- function (loc, filter_gene_name = NULL, filter_gene_biotype = NULL, 
    border = FALSE, cex.axis = 1, cex.lab = 1, cex.text = 0.7, 
    gene_col = ifelse(showExons, "blue4", "skyblue"), exon_col = "blue4", 
    exon_border = "blue4", showExons = TRUE, maxrows = NULL, 
    text_pos = "top", xticks = TRUE, xlab = NULL, highlight = "RGS14", 
    highlight_col = "#D7191C", blanks = c("fill", "hide")) 
{
    if (!inherits(loc, "locus")) 
        stop("Object of class 'locus' required")
    blanks <- match.arg(blanks)
    g <- locuszoomr::genetracks_grob(loc, filter_gene_name, filter_gene_biotype, 
        border, cex.text, gene_col, exon_col, exon_border, showExons, 
        maxrows, text_pos, highlight, highlight_col, blanks)
    if (is.null(xlab) & xticks) 
        xlab <- paste("Chromosome", loc$seqname, "(Mb)")
    g2 <- ggplot(data.frame(x = NA), aes(xmin = loc$xrange[1], xmax = loc$xrange[2])) + 
      #annotation_custom(grob = g)+
      scale_x_continuous(breaks = seq(ceiling(loc$xrange[1]/100000) * 100000,ceiling(loc$xrange[2]/100000) * 100000, by = 200000),
        labels = seq(ceiling(loc$xrange[1]/100000) * 100000,ceiling(loc$xrange[2]/100000) * 100000,by = 200000) / 1e+06 ,expand = c(0.01,0.01)) + 
      (if (!is.null(g)) gggrid::grid_panel(g)) + 
      scale_y_continuous(limits = c(0,1),breaks = seq(0,1,by=0.5),expand = c(0.01,0.03))+
      labs(x="Position on chr5 (Mb)",y="Gene")+
      theme(
        axis.line = element_line(color = "black"),   
        panel.background=element_rect(fill="white",color = "white"),
        panel.grid.major.y=element_blank(),
        panel.grid.major.x=element_blank(),
        legend.position = "none",
        legend.spacing.x = unit(4, "cm")
      )
}

ggsave(file = paste0("E://locus_zoom/test.pdf"), 
       plot = g2, 
       device = "pdf", 
       width = 10, 
       height = 5, 
       units = "cm")
```
![image](https://github.com/user-attachments/assets/2805f6b8-604c-4d84-8e41-6eed2211a177)

I have also tried reducing the `cex.text` argument to 0.3, but it did not resolve the issue. Some genes disappeared in the plot.
![image](https://github.com/user-attachments/assets/650b5338-d95f-450b-9e0c-ec3051e4d18d)

Thanks,
xicao

**Comment by Xi-Cao** [comment]
Hi again,

I realized that the disappearing genes were caused by my incorrect Y-axis setting. After correcting that, I was able to generate a complete image. However, I am still facing an issue where the gene text labels overlap with each other, even though I set `cex.text = 0.2`. I believe this is due to the limited number of rows and the close position between genes in one row of the plot. However, I couldn't find any arguments in the `genetracks_grob` function to modify these two aspects. It seems that there is only a argument `maxrow` to determine the max number of rows.

Could you please assist me in resolving this issue?

![image](https://github.com/user-attachments/assets/865f8f76-0fa9-44ca-b6dd-faf6d01c55ec)

Best,
xicao

**Comment by myles-lewis** [comment]
Hi xicao,

This is kind of a known/predictable issue. The text label overlap detection algorithm for grid grobs was always problematic at its core because it uses base graphics method for determining width of strings in plotting coordinates. This is particularly difficult to resolve because ggplot2 and grid graphics generate grobs which are replotted during the saving process. Once the gene track grob is generated the label positions will not update even if the width is modified by ggsave etc. That's what leads to the overlap. 

One possible solution is to fix the output width to 7 inch (that's the standard base graphics output window size) and scale everything else up accordingly to match such as a increase the height, font size etc.

Another solution would be to modify R base graphics default output window width to 10cm (3.94 inch) before creating the genetracks grob. That ought to stop the labels from overlapping:

```
dev.new(width=3.94)
```

Then call `dev.off()` after your gene tracks grob is created.

A 3rd solution is to switch to base graphics and avoid ggplot2 completely. The base graphics version determines the text overlap at the point of plotting/ exporting to pdf, so it will correctly avoid overlapping labels.

Bw, Myles

**Comment by Xi-Cao** [comment]
Hi Myles,

Thanks for your suggestions very much. Switching to base graphics effectively solved the problem. However, when I opted to use a larger graphic size with ggsave to address this, it became difficult to view after scaling it down to a normal size, as I need to stitch it with other ggplot images. Then, when I tried to scale everything else up accordingly (e.g., increasing the height, font size, etc.), the gene text overlapped again.

![image](https://github.com/user-attachments/assets/0a798cdb-d17a-4f29-b92e-112887ad33b3)

I've noticed that with base graphics, the row count is large enough to accommodate the genes and text without overlap. Would it be possible to control the row count of the grob in ggplot to ensure there is enough space for each gene and its corresponding text in each row?

![image](https://github.com/user-attachments/assets/83ef64a6-6980-4139-b876-8fb396e2a868)


Best,
xicao
